### PR TITLE
Issue #871: Fretboard: Update work manager implementation.

### LINKED ISSUE #871: Fretboard: Update work manager implementation
```
SyncWorker.kt: (10, 29): 'constructor Worker()' is deprecated. Deprecated in Java
WorkManagerSyncScheduler.kt: (34, 18): Parameter 'worker' is never used
```

### PR DESCRIPTION
**Pull request by fercarcedo**
Closes #871

### CODE REVIEW

File: components/service/fretboard/src/main/java/mozilla/components/service/fretboard/Fretboard.kt [hunk 1]
```diff
@@ -45,17 +45,19 @@ class Fretboard(
...
         }
         try {
             val serverExperiments = source.getExperiments(experimentsResult)
             experimentsResult = serverExperiments
             storage.save(serverExperiments)
+            return true
         } catch (e: ExperimentDownloadException) {
             // Keep using the local experiments
             logger.error(e.message, e)
+            return false
```
Comment (pocmo) [inline components/service/fretboard/src/main/java/mozilla/components/service/fretboard/Fretboard.kt]: nit you can move the return outside, AS may even recommend that:

```Kotlin
return try {
...
}
```

File: components/service/fretboard/src/main/java/mozilla/components/service/fretboard/scheduler/workmanager/WorkManagerSyncScheduler.kt [hunk 2]
```diff
@@ -35,7 +35,7 @@ class WorkManagerSyncScheduler {
         val constraints = Constraints.Builder()
             .setRequiredNetworkType(NetworkType.CONNECTED)
             .build()
-        val syncWork = PeriodicWorkRequest.Builder(SyncWorker::class.java, 1, TimeUnit.DAYS)
+        val syncWork = PeriodicWorkRequest.Builder(worker, 1, TimeUnit.DAYS)
             .setConstraints(constraints)
             .build()
         schedule(syncWork)
```
Comment (pocmo) [inline components/service/fretboard/src/main/java/mozilla/components/service/fretboard/scheduler/workmanager/WorkManagerSyncScheduler.kt]: nit: It looks like `schedule(request: WorkRequest)` doesn't need to be public. We may even be able to call `WorkManager.getInstance().enqueue()` here directly?

File: components/service/fretboard/src/main/java/mozilla/components/service/fretboard/scheduler/workmanager/WorkManagerSyncScheduler.kt [hunk 3]
```diff
@@ -35,7 +35,7 @@ class WorkManagerSyncScheduler {
         val constraints = Constraints.Builder()
             .setRequiredNetworkType(NetworkType.CONNECTED)
             .build()
-        val syncWork = PeriodicWorkRequest.Builder(SyncWorker::class.java, 1, TimeUnit.DAYS)
+        val syncWork = PeriodicWorkRequest.Builder(worker, 1, TimeUnit.DAYS)
```
Comment (pocmo) [inline components/service/fretboard/src/main/java/mozilla/components/service/fretboard/scheduler/workmanager/WorkManagerSyncScheduler.kt]: Maybe we can move the interval to be an optional parameter, like:

```Kotlin
    fun schedule(
        worker: Class<out SyncWorker>,
        interval: Pair<Long, TimeUnit> = Pair(1, TimeUnit.DAYS)
    ) {
```

File: components/service/fretboard/src/main/java/mozilla/components/service/fretboard/Fretboard.kt (hunk 1 above)
Comment (fercarcedo) [reply components/service/fretboard/src/main/java/mozilla/components/service/fretboard/Fretboard.kt]: Yes, I forgot about that!

File: components/service/fretboard/src/main/java/mozilla/components/service/fretboard/scheduler/workmanager/WorkManagerSyncScheduler.kt (hunk 2 above)
Comment (fercarcedo) [reply components/service/fretboard/src/main/java/mozilla/components/service/fretboard/scheduler/workmanager/WorkManagerSyncScheduler.kt]: Changed to private :)

File: components/service/fretboard/src/main/java/mozilla/components/service/fretboard/scheduler/workmanager/WorkManagerSyncScheduler.kt (hunk 3 above)
Comment (fercarcedo) [reply components/service/fretboard/src/main/java/mozilla/components/service/fretboard/scheduler/workmanager/WorkManagerSyncScheduler.kt]: Changed

**Comment by pocmo** [comment]
@fercarcedo It looks like you accidentally commited a bunch of `*.so` files. Can you update the PR? :)

**Comment by pocmo** [comment]
Thank you! No problem. I think this was caused by us listing and de-listing things in gitignore :)

**Pull request merged**